tests/utils/test_parallelize.py

```diff
@@ -51,54 +56,45 @@ def mock_runner(x, y, queue, func):
 @pytest.fixture(params=["numba_parallel", "numba_serial", "dask", "vanilla"])
 def func(request) -> Callable:
     return {
-        "numba_parallel": numba_parallel_func,
-        "numba_serial": numba_serial_func,
-        "dask": dask_func,
-        "vanilla": vanilla_func,
+        "numba_parallel": partial(wrap_numba_check, inner_function=numba_parallel_func),
+        "numba_serial": partial(wrap_numba_check, inner_function=numba_serial_func),
+        "dask": partial(wrap_numba_check, inner_function=dask_func),
+        "vanilla": partial(wrap_numba_check, inner_function=vanilla_func),
     }[request.param]
 
 
-@pytest.mark.timeout(60)
-@pytest.mark.parametrize("n_jobs", [1, 2, 8])
-def test_parallelize_loky(func, n_jobs):
-    start_time = time.time()
+# Timeouts are also useful because some processes don't return in
+# in case of failure.
+
+
+@pytest.mark.timeout(30)
+def test_parallelize_loky(func):
+    _test_parallelize(func, "loky")
+
+
+@pytest.mark.timeout(30)
+@pytest.mark.skipif(os.environ.get("CI") == "true", reason="Only testing 'loky' backend in CI environment")
+@pytest.mark.parametrize("backend", ["threading", "multiprocessing"])
+def test_parallelize_others(func, backend):
+    _test_parallelize(func, backend)
+
+
+def _test_parallelize(func, backend):
```

Comment:
```suggestion
@pytest.mark.timeout(30)
@pytest.mark.parametrize("backend", ["threading", pytest.param("multiprocessing", marks=@pytest.mark.skipif(os.environ.get("CI") == "true", reason="Only testing 'loky' backend in CI environment")), "loky"])
def test_parallelize(func, backend):
```

Or something to this effect.